src/components/TableDialog.vue

```diff
@@ -200,6 +203,15 @@ export default defineComponent({
     },
   },
   setup(props, { emit }) {
+    const { workspace } = props;
+
+    // Check for permissions
+    const hasPerms = ref(false);
+    watchEffect(async () => {
+      const perms = await api.getWorkspacePermissions(workspace);
+      hasPerms.value = canUpload(store.state.userInfo, perms);
+    });
```

Comment:
I believe there's a bug here, due to destructuring the `workspace` props. This kills the reactivity, so when you switch workspaces, this isn't updated. You should do the following instead:

```suggestion
    // Check for permissions
    const hasPerms = ref(false);
    watchEffect(async () => {
      const perms = await api.getWorkspacePermissions(props.workspace);
      hasPerms.value = canUpload(store.state.userInfo, perms);
    });
```